### ISSUE TITLE
Following build logs creates 1 socket/second & duplicate log data

### ISSUE THREAD

**Issue by jupierce** [openshift/jenkins-plugin#74]
If you create a start-build step & specify that build logs should be streamed, you will see in the Jenkins console that each build log message is appended to the console log multiple times. 

In IOpenShiftBuilder.java, you can see why:

```
        while (System.currentTimeMillis() < (startTime + wait)) {
            bld = client.get(ResourceKind.BUILD, bldId, getNamespace(overrides));
            bldState = bld.getStatus();
            if (Boolean.parseBoolean(getVerbose(overrides)))
                listener.getLogger().println("\nOpenShiftBuilder bld state:  " + bldState);

            if (follow) {
                final String container = pod.getContainers().iterator().next().getName();
                listener.getLogger().println("CONATINERNAME: " + container);

                stop = pod.accept(new CapabilityVisitor<IPodLogRetrievalAsync, IStoppable>() {
```

Each second, a new pod.accept is being invoked (which creates a net new socket/listener for the life of the build process). 

The simple fix is to move the `if (follow){...}` block outside of the loop. However, invoking .accept too early results in an exception which is currently swallowed by the restclient. 

```
Oct 05, 2016 6:12:57 PM hudson.init.impl.InstallUncaughtExceptionHandler$DefaultUncaughtExceptionHandler uncaughtException
SEVERE: A thread (OkHttp Dispatcher/95) died unexpectedly due to an uncaught exception, this may leave your Jenkins in a bad way and is usually indicative of a bug in the code.
com.openshift.restclient.BadRequestException: https://10.13.137.145:8443/api/v1/namespaces/myproject/pods/ruby-hello-world-8-build/log?container=docker-build&follow=true container "docker-build" in pod "ruby-hello-world-8-build" is waiting to start: ContainerCreating
    at com.openshift.internal.restclient.okhttp.ResponseCodeInterceptor.createOpenShiftException(ResponseCodeInterceptor.java:97)
    at com.openshift.internal.restclient.okhttp.ResponseCodeInterceptor.intercept(ResponseCodeInterceptor.java:59)
    at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
    at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
    at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:170)
    at okhttp3.RealCall.access$100(RealCall.java:33)
    at okhttp3.RealCall$AsyncCall.execute(RealCall.java:120)
    at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    at java.lang.Thread.run(Thread.java:745)
```

In other words, there is presently no (straightforward) way to know whether the operation fails & needs to be retried. 

Fixing it might require a slight API break in the openshift-restclient, so I've opened a PR to debate the discuss the change: openshift/openshift-restclient-java#224

**Comment by gabemontero** [comment]
Hmmm ... i thought @jim-minter fixed duplicate log data with https://github.com/openshift/jenkins-plugin/commit/7da55034807b98c66f897cbca7fddf7865fa8ef2 - do you have this commit in your env ?

**Comment by jupierce** [comment]
Ah, no I don't. That will work in the vast majority of runs, but it is still non-deterministic because it relies on a momentary build state.

**Comment by gabemontero** [comment]
OK, given @jim-minter 's change covers the vast majority, I've marked a p3.

As an fyi, up until a couple of months ago, the restclient's pod logging depended on the `oc` binary being present, so we had our own polling mechanism that did simply http requests against the build log endpoint, and we could manage the edge cases more directly.

So let's see how the discussion you started goes.  If it doesn't go the way we want it, we can always revert to our former bypass path if it ends up cleaner than the retry approach with the Pod logging @jupierce noted.

@jupierce and I have talked, and given the significant amount of leg work he's already done, he's agreed to take ownership of this one.  Thanks!
